### PR TITLE
test: More Page Models for Experiment Tracking [INFENG-686]

### PR DESCRIPTION
**Pull request by JComins000**
## Ticket

[INFENG-686]

## Description

* the dropdown refactor really helped with all this action bar stuff.
* testids
* I want to make readme changes and global docstring changes in another PR. I think, in general, if a class isn't getting exported, it might be okay to be lazy about the docstring. I'd be open to suggestions here. https://hpe-aiatscale.atlassian.net/browse/INFENG-696

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[INFENG-694]: https://hpe-aiatscale.atlassian.net/browse/INFENG-694?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[INFENG-686]: https://hpe-aiatscale.atlassian.net/browse/INFENG-686?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keita-determined — APPROVED** [review]

File: webui/react/src/e2e/models/components/FilterForm/components/FilterGroup.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * This constructor represents the contents in src/components/FilterGroup.tsx.
+ * @param {object} obj
+ * @param {CanBeParent} obj.parent - The parent used to locate this FilterGroup
+ * @param {string} obj.selector - Used instead of `defaultSelector`
+ */
+export class FilterGroup extends NamedComponent {
+  readonly defaultSelector = '[data-test-component="FilterGroup"]';
+
+  #childrenSelector = '[data-test="children"]';
+  #notNestedSelector = `:not(${this.#childrenSelector} *)`;
```
Comment (djanicekpach) [inline webui/react/src/e2e/models/components/FilterForm/components/FilterGroup.ts:17]: should these be `private`/`readonly`?

**Review by djanicekpach — APPROVED** [review]

File: webui/react/src/components/FilterForm/components/FilterField.tsx [hunk 2]
```diff
@@ -187,9 +187,10 @@ const FilterField = ({
           formStore.setFieldConjunction(parentId, (value?.toString() ?? 'and') as Conjunction);
         }}
       />
-      <div className={css.fieldCard} ref={preview}>
+      <div className={css.fieldCard} data-test="fieldCard" ref={preview}>
         <Select
           autoFocus
+          data-test="columnName"
```
[the remark this answers is not in the document]
Comment (JComins000) [reply webui/react/src/components/FilterForm/components/FilterField.tsx:193]: yeah i wanted to add this to the readme.

i'll put it in my readme.
data-test-component -> top level id for components
data-test -> test hooks for sub components
data-testid -> i'm using it for the first time here. it's for dynamic data

As an example, ColumnPickerTab gets `data-test-component="columnPickerTab"` at it's top level. rows inside the column picker all have `data-test="row"` . Each row will have a `data-test-id="${testid}"`, which is equal to `col.column` in production code. It's just the name of the column. It's much easier than grabbing, say, the nth row.

[the remark this answers is not in the document]
Comment (JComins000) [reply webui/react/src/components/FilterForm/components/FilterField.tsx:193]: it's possible to use all three keys in one element. that's why we need many

File: webui/react/src/e2e/models/components/FilterForm/components/FilterGroup.ts (hunk 1 above)
Comment (JComins000) [reply webui/react/src/e2e/models/components/FilterForm/components/FilterGroup.ts:17]: yes, let me add readonly. theyre already private

**Pull request merged**